inference-engine/tests/functional/plugin/myriad/ngraph/operations/dynamic_shape_resolver.cpp

```diff
@@ -41,8 +41,9 @@ class DynamicShapeResolverTests : public CommonTestUtils::TestsCommon, public te
 
 TEST_P(DynamicShapeResolverTests, CanValidateAndInferTypes) {
     std::shared_ptr<ngraph::vpu::op::DynamicShapeResolver> dynamicShapeResolver;
+    std::shared_ptr<ngraph::Function> fun;
     ASSERT_NO_THROW(dynamicShapeResolver = std::make_shared<ngraph::vpu::op::DynamicShapeResolver>(data, dims));
-    ASSERT_NO_THROW(std::make_shared<ngraph::Function>(ngraph::NodeVector{dynamicShapeResolver}, ngraph::ParameterVector{data, dims}));
+    ASSERT_NO_THROW(fun = std::make_shared<ngraph::Function>(ngraph::NodeVector{dynamicShapeResolver}, ngraph::ParameterVector{data, dims}));
```

Comment:
Here and in other places: what is the subject of the warning? Unused returned object? If so, it seems strange since it's completely ok for C++ code.

Anyway, don't see a reason to instantiate an object outside expression if it's unused. Could you please try something similar to following:

```suggestion
    ASSERT_NO_THROW(auto function = std::make_shared<ngraph::Function>(ngraph::NodeVector{dynamicShapeResolver}, ngraph::ParameterVector{data, dims}));
```

Reply:
warning C4834: discarding return value of function with 'nodiscard' attribute

Reply:
But `nodiscard` attribute is introduced since C++ 17 only, while we use C++ 11.

Reply:
MSVC compiler actually enables features if they available. This place was spotted by warning, can share my log if you interesting

Reply:
Got it, please take into consideration my suggestion regarding fix if possible